### PR TITLE
Rationalize use of quotation marks

### LINKED ISSUE #1011: Use single quotes consistently in messages
Sometime we use double quotes, sometimes single, sometimes none. Let's make this consistent.

See https://github.com/FakeItEasy/FakeItEasy/pull/1001/files#r104390257

### PR DESCRIPTION
**Pull request by blairconrad**
Fixes #1011:
- double quotes for string values
- no quotes when values cannot have spaces
- single quotes otherwise

### CODE REVIEW

File: src/FakeItEasy/Core/DefaultFixtureInitializer.cs [hunk 1]
```diff
@@ -31,7 +31,7 @@ private static SettableMemberInfo GetSutSetter(object fixture)
 
             if (CountIsNoMoreThanOne(allSettersTaggedUnderTest))
             {
-                throw new InvalidOperationException("A fake fixture can only contain one member marked \"under test\".");
+                throw new InvalidOperationException($"A fake fixture can only contain one member marked with {nameof(UnderTestAttribute)}.");
```
Comment (blairconrad) [inline src/FakeItEasy/Core/DefaultFixtureInitializer.cs]: This seemed more clear.

File: tests/FakeItEasy.Specs/SelfInitializedFakesSpecs.cs [hunk 2]
```diff
@@ -218,7 +218,7 @@ public static class SelfInitializedFakesSpecs
             // These results demonstrate that the self-initialized fake relies on a script
             // defined by which methods are called, without regard to the arguments
             // passed to the methods.
-            "And the playback fake returns results in 'recorded order'"
+            "And the playback fake returns results in recorded order"
```
Comment (blairconrad) [inline tests/FakeItEasy.Specs/SelfInitializedFakesSpecs.cs]: I didn't see what value the quotes provided.

File: src/FakeItEasy/Configuration/PropertyExpressionHelper.cs [hunk 3]
```diff
@@ -39,7 +39,7 @@ internal class PropertyExpressionHelper
                 }
 
                 throw new ArgumentException(
-                    "The property '" + propertyName + "' does not have a setter.");
+                    "The property " + propertyName + " does not have a setter.");
```
Comment (thomaslevesque) [inline src/FakeItEasy/Configuration/PropertyExpressionHelper.cs]: string interpolation would probably be more readable here

File: src/FakeItEasy/Creation/CastleDynamicProxy/DynamicProxyMessages.cs [hunk 4]
```diff
@@ -11,7 +11,7 @@ internal static class DynamicProxyMessages
             "Arguments for constructor specified for interface type.";
 
         public static string ProxyIsSealedType(Type type) =>
-            $@"The type of proxy ""{type}"" is sealed.";
+            $@"The type of proxy {type} is sealed.";
```
Comment (thomaslevesque) [inline src/FakeItEasy/Creation/CastleDynamicProxy/DynamicProxyMessages.cs]: The `@` is no longer needed

File: tests/FakeItEasy.Tests/NullGuardedAssertion.cs [hunk 5]
```diff
@@ -292,7 +292,7 @@ private void WriteFailReason(StringBuilderOutputWriter description)
                             if (argumentNullException != null)
                             {
                                 description.Write(
-                                    $@"threw ArgumentNullException with wrong argument name, it should be ""{this.ArgumentName}"".");
+                                    $@"threw ArgumentNullException with wrong argument name, it should be {this.ArgumentName}.");
```
Comment (thomaslevesque) [inline tests/FakeItEasy.Tests/NullGuardedAssertion.cs]: `@` no longer needed

**Comment by blairconrad** [comment]
Thanks, @thomaslevesque. I feel silly for not getting rid of the `@`s!

Don't merge! It's not squashed. :wink:

**Comment by thomaslevesque** [comment]
LGTM, I'll merge after you squash

**Comment by blairconrad** [comment]
Thanks, @thomaslevesque. Squashed.

**Comment by thomaslevesque** [comment]
Thanks @blairconrad

**Pull request merged**